### PR TITLE
Adding range comparision functionality to aggdq Detailed Stats table

### PR DESCRIPTION
**Pull request by nishantsingh93**
## Description

This change introduces range comparison for Aggregation Data Quality expectation. A user can now provide valid SQL expression to compare upper_bound and lower_bound check against a measure 
## Related Issue

https://github.com/Nike-Inc/spark-expectations/issues/81

## Motivation and Context

This feature request is to enhance the Aggregation Data quality to accept a range expression. There are two types of valid SQL expression that are supported. 
* `between lower_bound and upper_bound`
* `>lower_bound and <upper_bound`
### Example:
1) sum(sales) between 1000 and 10000
2) sum(sales)>1000 and sum(sales)<10000
## How Has This Been Tested?

* Added new test cases to actions module
* Added new test cases to actions test_expectations module
*  Added all the Docs to the md file

* Ran the test in local using `sample_dq_iceberg.py` and `sample_dq_delta.py`

* This change affects detailed stats table when enabled `true` and does not touch any other functionalities of the code
## Screenshots (if appropriate):
Screenshot 2024-04-09 at 12 06 51 AM
Screenshot 2024-04-09 at 12 07 23 AM
Screenshot 2024-04-09 at 12 07 37 AM
Screenshot 2024-04-09 at 12 07 47 AM

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: spark_expectations/notifications/push/spark_expectations_notify.py [hunk 1]
```diff
@@ -41,6 +41,8 @@ def wrapper(*args: List, **kwargs: Dict) -> DataFrame:
                     # self._context.set_dq_start_time()
                     result = func(*args, **kwargs)
 
+                    print("verfing")
```
Comment (asingamaneni) [inline spark_expectations/notifications/push/spark_expectations_notify.py]: should we remove this?

Comment (nishantsingh93) [reply spark_expectations/notifications/push/spark_expectations_notify.py]: removed!!

File: docs/configurations/rules.md
Comment (jskrajareddy21) [inline docs/configurations/rules.md:1]: ```count(*) >[lower_bound] and < [upper_bound]``` add count before upper bound

**Review by asingamaneni — APPROVED** [review]
LGTM, thanks for the contribution

**Pull request merged**